docs/CloudStorageToSnowflake-action.md

```diff
@@ -0,0 +1,183 @@
+
+**Master Key:** Client-side master key that Snowflake uses to encrypt the files containing the 
+unloaded data. The master key must be a 128-bit or 256-bit key in Base64-encoded form. Required by
+`AWS_CSE` and `AZURE_CSE` encryption types.
+
+**KMS Key Id:** ID for the KMS-managed key that is used to encrypt files unloaded into the bucket. 
+If no value is provided, your default KMS key ID is used to encrypt files on unload. The value is optionally
+and used by the following encryption types: `AWS_SSE_KMS`, `GCS_SSE_KMS`.
+
+### File Format
+
+**File Format Filtering Policy:** Policy used to filter source files. Possible value:
+- `Undefined` Do not filter the files.
+- `By File Type` Filter using `Format Type` configuration.
+- `By Existing Format Specification` Filter by existing format saved in Snowflake.
+
+**Format Name:** Existing named file format to use for loading data into the table. The named file 
+format determines the format type (CSV, JSON, etc.), as well as any other format options, for the data files. 
+For more information, see [CREATE FILE FORMAT.](https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html)
+
+**Format Type:** Type of files to load into the table. Supported values; CSV, JSON, AVRO, ORC, PARQUET, XML.
+
+**Format Type Options:** Depending on the file format type specified (`Format Type` property) 
+you can include one or more of the format-specific options. Find the options list 
+[here.](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#format-type-options-formattypeoptions).
+
+### Advanced
+
+**Files:** List of one or more files names (separated by commas) to be loaded. The files must already have 
+been staged in either the Snowflake internal location or external location specified in the command.
+The maximum number of files names that can be specified is 1000.
+           
+Note. For external stages only (Amazon S3, Google Cloud Storage, or Microsoft Azure), the file path is set by concatenating 
+the URL in the stage definition and the list of resolved file names.
+However, Snowflake doesn’t insert a separator implicitly between the path and file names. 
+You must explicitly include a separator (/) either at the end of the URL in the stage definition or 
+at the beginning of each file name specified in this parameter.
+
+**Pattern:** Regular expression pattern string, enclosed in single quotes, specifying the file names 
+and/or paths to match. 
+
+Tip. For the best performance, try to avoid applying patterns that filter on a large number of files.
+
+**Copy Options:** List of copy options. Find them 
+[here.](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#copy-options-copyoptions)
+
+**Connection Arguments:** List of arbitrary string tag/value pairs as connection arguments. See: [JDBC Driver Connection String.](https://docs.snowflake.com/en/user-guide/jdbc-configure.html#jdbc-driver-connection-string)
```

Comment:
To use this plugin, does user need to deploy jdbc driver on data fusion?

Reply:
No. We get jdbc lib via maven.

Reply:
what is the jdbc license? is it open source?

Reply:
@pgancz can you please answer here.

Reply:
Yes the JDBC driver is open source.  The licenses, as well as the source, can be found here - https://repo1.maven.org/maven2/net/snowflake/snowflake-jdbc/3.12.5.  (Licenses.html is in the javadoc jar file).